### PR TITLE
🧹 Cleanup some `TODOs`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
- Drop deletion of deprecated `allow-to-shoot-networks` `NetworkPolicy` (part of https://github.com/gardener/gardener/pull/9752, released with `v1.96.0`)
- Drop fetching extension observability configs with deprecated/legacy method (part of https://github.com/gardener/gardener/pull/9695, released with `v1.95.0`)
- Drop Prometheus/Alertmanager migration coding (part of https://github.com/gardener/gardener/pull/9695, released with `v1.95.0`)
- Drop deprecated `.spec.pools[].userData` from `extensions.gardener.cloud/v1alpha1.Worker` API (part of https://github.com/gardener/gardener/pull/9722, released with `v1.95.0`)
- Drop OSC hash migration `Secret` creation (part of https://github.com/gardener/gardener/pull/9846, released with `v1.97.0`)
- Drop OSC hash assertion from upgrade tests (part of https://github.com/gardener/gardener/pull/9865, released with `v1.98.0`)
- Drop removal code of `HVPA` resources (part of https://github.com/gardener/gardener/pull/9698, released with `v1.95.0`)

/cc @ScheererJ @MichaelEischer 
FYI @istvanballok @andrerun 

**Release note**:

```breaking developer
The legacy method of providing monitoring configuration via `ConfigMap`s labeled with `extensions.gardener.cloud/configuration=monitoring` has been removed. See [this](https://github.com/gardener/gardener/blob/master/docs/extensions/logging-and-monitoring.md#monitoring) instead.
```

### CODE REVIEW

**Review by ScheererJ — COMMENTED** [review]
Thanks a lot for cleaning up 🧹.

/lgtm

**Comment by ialidzhikov** [comment]
Sorry for the delay on my review, I am now checking the PR.

File: docs/extensions/logging-and-monitoring.md [hunk 1]
```diff
@@ -319,78 +319,6 @@ data:
   my-custom-dashboard.json: <dashboard-JSON-document>
 ```
 
-#### Legacy Method of Providing Observability Configuration
```
Comment (ialidzhikov) [inline docs/extensions/logging-and-monitoring.md:322]: FYI @vpnachev there is still no release of the shoot-lakom-service that includes https://github.com/gardener/gardener-extension-shoot-lakom-service/pull/87. I assume this cleanup commit will break the monitoring config provided by the extension. You may want to kick off a new release process for the shoot-lakom-service to prevent that.

File: pkg/component/shared/prometheus.go [hunk 2]
```diff
@@ -23,27 +23,5 @@ func NewPrometheus(log logr.Logger, c client.Client, namespace string, values pr
 	values.Image = imagePrometheus.String()
 	values.Version = ptr.Deref(imagePrometheus.Version, "v0.0.0")
 
-	// TODO(rfranzke): Remove this block after v1.97 has been released.
-	{
-		imageAlpine, err := imagevector.Containers().FindImage(imagevector.ContainerImageNameAlpine)
```
Comment (ialidzhikov) [inline pkg/component/shared/prometheus.go:28]: This image can be now removed as it is no longer referenced: https://github.com/gardener/gardener/blob/bd81ac3afcb4a30cfe5e5e047d7ada09742ce616/imagevector/images.go#L12-L13

**Review by ialidzhikov — COMMENTED** [review]
/lgtm
code-wise but I am not sure about https://github.com/gardener/gardener/pull/10220#discussion_r1697977594, whether we should proceed with having in mind that it will break an extension. Maybe @vpnachev can share whether he will make it in 1-2 weeks to cut a new shoot-lakom-service release?

**Comment by rfranzke** [comment]
> /lgtm code-wise but I am not sure about [#10220 (comment)](https://github.com/gardener/gardener/pull/10220#discussion_r1697977594), whether we should proceed with having in mind that it will break an extension. Maybe @vpnachev can share whether he will make it in 1-2 weeks to cut a new shoot-lakom-service release?

We had such cases in the past. I don't think this model scales (= we/core checks if all extensions have been adapted accordingly, otherwise core is blocked with moving forward). It does not even work with extensions outside of this GitHub org which we are not aware of. Yes, we can consider extensions in this GitHub org as "special", but I don't think it's a very neat approach.

We already use 6 releases (=12 weeks) of deprecation before we finally remove something that could affect extensions. This should be enough time to click the release buttons. If all this doesn't work, we should rather consider moving extensions back in-tree to have the control and flexibility we want. 

TL;DR: I propose @vpnachev uses this and next week (before `v1.101` gets released) to click the release button, such that we can proceed with this PR and the removal timelines.

**Comment by vpnachev** [comment]
Thanks for the heads up!
I was planning to release a new version of lakom after we implement one improvement in the following weeks, but obviously now the plan will be changed and I will do it now.  

> We already use 6 releases (=12 weeks) of deprecation before we finally remove something that could affect extensions. This should be enough time to click the release buttons.

You are right, however components that can be considered feature complete and not under active development should not be expected to be released often.

**Review by ialidzhikov — COMMENTED** [review]
> I was planning to release a new version of lakom after we implement one improvement in the following weeks, but obviously now the plan will be changed and I will do it now.

Thanks! I am approving then
/approve

**Comment by rfranzke** [comment]
Thanks colleagues, appreciated!
